### PR TITLE
3.24.x deps 2024 09 06 13 39 17 290986

### PR DESCRIPTION
**Pull request by craigcomstock**

### CODE REVIEW

**Comment by craigcomstock** [comment]
@cf-bottom jenkins no test

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11186/)

 [NO TESTS]

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11186/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11186/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins no tests

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11192/)

 [NO TESTS]

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11192/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11192/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins, think I'm ready for the tests

**Comment by craigcomstock** [comment]
Oops. Forgot a fix somewhere from cf-bottom not quite working.
```
10:33:40 error: File /home/jenkins/workspace/testing-pr/label/PACKAGES_x86_64_linux_redhat_7/openssl/SOURCES/openssl-3.3.1.tar.gz: No such file or directory
```
will fix.

**Comment by craigcomstock** [comment]
@cf-bottom jenkins

**Comment by craigcomstock** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11202/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11202/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11202/

**Comment by craigcomstock** [comment]
@vpodzime as you can see looking back in comments here we have a green jenkins build above with these changes.

I just squashed the two fixup commits for openssl into one commit with the upgrade.

Ready for review.

**Review by larsewi — APPROVED** [review]

**Comment by larsewi** [comment]
Maybe you can update the comment above

> Updated postgresql from 16.3 to 15.8

I got a bit confused at first ;)

File: build-scripts/install-dependencies [hunk 1]
```diff
@@ -40,13 +40,20 @@ check_and_install_perl()
...
     then
         echo "$PERL version is $PERL_MINOR_VERSION," \
-             "too old, installing new one from source"
-
-        ### BUILD PERL
+             "too old"
+        PERL_OK="no"
+    fi
+    if ! perl -e 'use List::Util qw(pairs);'; then
+        echo "$PERL has List::Util that does not export pairs. Needs to be at least version 1.29 for OpenSSL version 3.3.2."
```
Comment (vpodzime) [inline build-scripts/install-dependencies:52]: Hmm, AFAICT we check that Perl is >= 1.30 above. So this seems wrong.

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise. Please check things work not just build on exotics (unless you already did).

File: build-scripts/install-dependencies (hunk 1 above)
Comment (craigcomstock) [reply build-scripts/install-dependencies:52]: This is List::Util version 1.29 or above.

The Perl version is not sufficient to check this particular aspect. We probably shouldn't have relied on a Perl version in the first place and instead put in exactly what was needed like this List::Util exporting pairs.

**Comment by craigcomstock** [comment]
> Maybe you can update the comment above
> 
> > Updated postgresql from 16.3 to 15.8
> 
> I got a bit confused at first ;)

Funny, that's the initial comment that github makes from the initial PR. That wrong version was generated by cf-bottom and then I fixed it. The commit comment is correct and now I have fixed the PR comment. :+1: Good catch.

**Comment by craigcomstock** [comment]
> Looks good to me otherwise. Please check things work not just build on exotics (unless you already did).

Right. I will build and test on exotics.

**Comment by craigcomstock** [comment]
check aix71 since it was offline
[Build Status](https://ci.cfengine.com/job/pr-pipeline/11214/)

**Comment by olehermanse** [comment]
@craigcomstock why does PR title say 3.24.x but target branch is master?

**Comment by craigcomstock** [comment]
> @craigcomstock please update the tables in the README (Version numbers in master column and add 3.24.x column).

Good catch. I used cf-bottom to start the PR but it did a bad job due to a few deps not working well for various reasons.

I will add a todo to the ticket to check this in all three branches: 3.21.x, 3.24.x and master.

**Comment by craigcomstock** [comment]
> @craigcomstock why does PR title say 3.24.x but target branch is master?

No idea. I am not actively working on this PR right now, but rather the master one here: https://github.com/cfengine/buildscripts/pull/1493

So I will fix things up in due course.

**Comment by olehermanse** [comment]
> I will add a todo to the ticket to check this in all three branches: 3.21.x, 3.24.x and master.

@craigcomstock I believe we've just updated the table on the master branch in the past (with correct versions from the other LTS branches).

**Comment by craigcomstock** [comment]
reworked the commits from the master deps PR https://github.com/cfengine/buildscripts/pull/1493

**Comment by craigcomstock** [comment]
Added cherry pick of 3.21.x removal of dependency table to this PR since that will go inline with this change as well.

@cf-bottom jenkins with exotics

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11221/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11221/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11221/

**Comment by craigcomstock** [comment]
retry hpux no-test job: [Build Status](https://ci.cfengine.com/job/pr-pipeline/11224/)

**Comment by craigcomstock** [comment]
bootstrap-tests, restart failure is a flake on exotics where cf-serverd is not running, ENT-9767
sequential-tests two failures
- no-noise test, centos-7, ENT-12191
- cf-support on hpux, ENT-12192

File: deps-packaging/openssl/source [hunk 2]
```diff
@@ -1 +1 @@
-https://www.openssl.org/source/
+https://github.com/openssl/openssl/releases/download/openssl-3.3.2/
```
Comment (vpodzime) [inline deps-packaging/openssl/source:1]: This change of the source is worth a comment in the commit message. Also, the commit message has a `cherry-picked from` reference to your private branch.

File: build-scripts/install-dependencies [hunk 3]
```diff
@@ -78,12 +78,14 @@ check_and_install_perl()
...
+        PERL_MD5=8da78b1f54b99e97954066d0aaad88bc
+        wget http://www.cpan.org/src/5.0/perl-${PERL_VERSION}.tar.gz
+        [ `func_md5 perl-${PERL_VERSION}.tar.gz` != "${PERL_MD5}" ]  \
             &&  fatal "perl checksum error"
-        gzip -dc perl-5.26.1.tar.gz | tar xf -
-        cd perl-5.26.1
-        $PATCH -p1 < $BASEDIR/buildscripts/build-scripts/perl-488307ffa6.patch
+        gzip -dc perl-${PERL_VERSION}.tar.gz | tar xf -
+        cd perl-${PERL_VERSION}
+#        $PATCH -p1 < $BASEDIR/buildscripts/build-scripts/perl-488307ffa6.patch
```
Comment (vpodzime) [inline build-scripts/install-dependencies:88]: Time to drop this patching and the patch itself?

**Review by vpodzime — COMMENTED** [review]
I think the last commit should be part of the OpenSSL version bump. Also please remove the `cherry picked` references to private branches/commits.

File: build-scripts/install-dependencies (hunk 3 above)
Comment (craigcomstock) [reply build-scripts/install-dependencies:88]: yeah, probably so. :+1:

**Comment by craigcomstock** [comment]
> I think the last commit should be part of the OpenSSL version bump. Also please remove the `cherry picked` references to private branches/commits.

There are 4 commits related to OpenSSL version bump. Do you think all of them should be in one?

https://github.com/cfengine/buildscripts/pull/1494/commits/e3dc8b5beff7b6ada529b4183635596e76eca0db

https://github.com/cfengine/buildscripts/pull/1494/commits/d460ee01621cdd7abebb1c0e4da39ac38550491d

https://github.com/cfengine/buildscripts/pull/1494/commits/9910ebc4ad7aa7e19e2177e5ec88a647f65d224b

https://github.com/cfengine/buildscripts/pull/1494/commits/d4a1cb14f8c4bdb848b871edf7e6f145a31fee0e

File: deps-packaging/openssl/source (hunk 2 above)
Comment (craigcomstock) [reply deps-packaging/openssl/source:1]: FWIW https://openssl-library.org/source/ links every release to github.com, there are no releases and openssl.org/source/ anymore.

**Comment by craigcomstock** [comment]
This PR is proposed to merge to master. This is incorrect for this 3.24.x deps update. Will resubmit a new PR.

**Pull request closed without merging**